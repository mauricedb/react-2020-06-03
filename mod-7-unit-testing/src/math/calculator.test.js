import Calculator from "./Calculator";

function wait() {
  return new Promise(function (resolve, reject) {
    setTimeout(function () {
      resolve();
    }, 1000);
  });
}

describe("the calculator", function () {
  let calc;

  beforeAll(function () {
    calc = new Calculator();
  });
  //   beforeEach(function () {
  //     calc = new Calculator();
  //   });

  describe("can add", function () {
    test("2 + 5 === 7", function () {
      const result = calc.add(2, 5);

      expect(result).toBe(7);
    });

    test("async", function () {
      return wait();
    });

    test("async", async function () {
      await wait();
    });

    test("1 + 1 === 2", function () {
      const result = calc.add(1, 1);

      expect(result).toBe(2);
    });

    test("0.1 + 0.2 === 0.3", function () {
      const result = calc.add(0.1, 0.2);

      expect(result).toBeCloseTo(0.3);
    });
  });

  describe("can subtract", function () {
    test("2 - 5 === -3", function () {
      const result = calc.subtract(2, 5);

      expect(result).toBe(-3);
    });

    test("1 - 1 === 0", function () {
      const result = calc.subtract(1, 1);

      expect(result).toBe(0);
    });

    test("1 - 1 === 0", function () {
      const result = calc.subtract(1, 1);

      expect([{ x: 1, name: "John" }]).toMatchSnapshot();
    });
  });
});
