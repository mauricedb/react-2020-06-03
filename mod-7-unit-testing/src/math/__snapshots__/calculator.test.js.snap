// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the calculator can subtract 1 - 1 === 0 1`] = `
Array [
  Object {
    "name": "Jack",
    "x": 1,
  },
]
`;
